ArcGISRuntimeSDKQt_CppSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.cpp

```diff
@@ -0,0 +1,226 @@
+  QObject(parent),
+  m_portalItem(new PortalItem("acc027394bc84c2fb04d1ed317aac674", this)),
+  m_lineSymbol(new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor(Qt::red), 5, this)),
+  m_graphicParent(new QObject())
+{
+  m_busy = true;
+  emit busyChanged();
+
+  m_graphicsOverlay = new GraphicsOverlay(this);
+  m_graphicsOverlay->setRenderer(new SimpleRenderer(m_lineSymbol, this));
+
+  connect(m_portalItem, &PortalItem::doneLoading, this, [this] ()
+  {
+    m_offlineMapTask = new OfflineMapTask(m_map, this);
+    m_mapView->graphicsOverlays()->append(m_graphicsOverlay);
+
+    connect(m_offlineMapTask, &OfflineMapTask::doneLoading, this, [this] ()
+    {
+      connect(m_offlineMapTask, &OfflineMapTask::preplannedMapAreasCompleted, this, &DownloadPreplannedMap::loadPreplannedMapAreas);
+      m_offlineMapTask->preplannedMapAreas();
+    });
+
+    m_offlineMapTask->load();
+  });
+
+  m_map = new Map(m_portalItem, this);
+}
+
+DownloadPreplannedMap::~DownloadPreplannedMap() = default;
+
+void DownloadPreplannedMap::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DownloadPreplannedMap>("Esri.Samples", 1, 0, "DownloadPreplannedMapSample");
+}
+
+MapQuickView* DownloadPreplannedMap::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void DownloadPreplannedMap::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  emit mapViewChanged();
+}
+
+void DownloadPreplannedMap::downloadMapArea(int index)
+{
+  if (!m_offlineMapTask)
+    return;
+
+  if (m_offlineMapTask->preplannedMapAreaList()->isEmpty())
+    return;
+
+  if (m_offlineMapTask->preplannedMapAreaList()->at(index)->loadStatus() != LoadStatus::Loaded)
+    return;
+
+  m_busy = true;
+  emit busyChanged();
+
+  QString path = QString(m_tempPath.path() + "/" + m_offlineMapTask->preplannedMapAreaList()->at(index)->portalItem()->title() + ".mmpk");
+
+  QDir tempDir(path);
+
+  if (m_mapExists)
+  {
+    if (m_mmpk)
+    {
+      delete m_mmpk;
+      m_mmpk = nullptr;
+    }
+
+    m_mmpk = new MobileMapPackage(path, this);
+
+    connect(m_mmpk, &MobileMapPackage::doneLoading, this, [this] (Error e)
+    {
+      if (!e.isEmpty())
+        qDebug() << e.message() << " - " << e.additionalMessage();
+
+      m_mapView->setMap(m_mmpk->maps().at(0));
+      m_busy = false;
+      emit busyChanged();
+      m_graphicsOverlay->setVisible(false);
+      m_viewingOnlineMaps = false;
+      emit viewingOnlineMapsChanged();
+    });
+
+    m_mmpk->load();
+    return;
+  }
+  else
+  {
+    connect(m_offlineMapTask, &OfflineMapTask::createDefaultDownloadPreplannedOfflineMapParametersCompleted, this, [this, path] (QUuid ,const DownloadPreplannedOfflineMapParameters &parameters)
+    {
+
+      m_params = DownloadPreplannedOfflineMapParameters(parameters);
```

Comment:
I think you should just be able to do:
```suggestion
      m_params = parameters;
```